tests/yam/agama/boot_agama.pm

```diff
@@ -55,9 +55,9 @@ sub run {
 
     # prepare kernel parameters
     if (my $agama_auto = get_var('AGAMA_AUTO')) {
-        my $path = expand_agama_profile($agama_auto);
-        set_var('AGAMA_AUTO', $path);
-        set_var('EXTRABOOTPARAMS', get_var('EXTRABOOTPARAMS', '') . " agama.auto=\"$path\"");
+        my $profile_url = ($agama_auto =~ /libsonnet/) ? generate_json_profile() : expand_agama_profile($agama_auto);
```

Comment:
better like an extension with `/.libsonnet/ but maybe you need to escape it with `/\.libsonnet/ I don't remember.

Reply:
when using ternary is much more clear use a line for each branch, so it will admit comments on top to clarify. Could you please move `generate_json_profile() to the next line?